api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -210,12 +154,24 @@ default Iterable<DeleteFile> removedDeleteFiles(FileIO io) {
   }
 
   /**
-   * Return the location of this snapshot's manifest list, or null if it is not separate.
+   * Return the location of this snapshot's manifest list, or null if it is not separate. If this
+   * method returns null, then {@link Snapshot#v1ManifestLocations()} must return non-null.
    *
    * @return the location of the manifest list for this Snapshot
    */
   String manifestListLocation();
 
+  /**
+   * Returns a collection of different manifest locations for this Snapshot, or null if manifests
+   * are tracked in a single manifest list file (thus {@link Snapshot#manifestListLocation()} must
+   * return non-null).
+   *
+   * @return A collection of different manifest locations for this Snapshot
+   */
+  default List<String> v1ManifestLocations() {
```

Comment:
I don't see a reason to add this method to the API. The existing methods should be fine for returning these. It doesn't matter where the manifests are tracked to callers.

Looks like this is only used for writing manifests in the metadata file as paths, which can be done using `allManifests`.

Reply:
As discussed in https://github.com/apache/iceberg/pull/5773#discussion_r978390263 I removed this method again from the `Snapshot` API